jupyterhub/user.py

```diff
@@ -336,6 +344,69 @@ def sync_groups(self, group_names):
             self.orm_user.groups = []
         self.db.commit()
 
+    def sync_roles(self, auth_roles):
+        """Synchronize roles with database"""
+        current_roles = {g.name for g in self.db.query(orm.Role).all()}
+
+        auth_roles_names = _get_role_names(auth_roles)
+
+        new_roles_names = set(auth_roles_names).difference(current_roles)
+        if new_roles_names:
+            self.log.info(f"Adding new roles {new_roles_names} to the database")
+        if auth_roles:
+            roles = self.db.query(orm.Role).all()
+            existing_roles = {g.name for g in roles}
+            for role_name in auth_roles_names:
+                if role_name not in existing_roles:
+                    # create roles that don't exist yet
+                    self.log.info(f"Creating new role {role_name}")
+
+                    role = _get_role_object(auth_roles, role_name)
+
+                    role_name = []
+                    if 'name' in role.keys():
+                        role_name = role['name']
```

Comment:
This is spurious. `role_name ` is already defined (this is what we iterate over), and `_get_role_object` would not return the role if the `name` would not exist.